src/main/scala/org/broadinstitute/dsde/agora/server/webservice/AgoraService.scala

```diff
@@ -108,27 +108,29 @@ abstract class AgoraService(permissionsDataSource: PermissionsDataSource) extend
   // GET http://root.com/configurations/<namespace>/<name>/<snapshotId>
   def querySingleRoute =
     matchQuerySingleRoute(path) { (namespace, name, snapshotId, username) =>
-      extractOnlyPayloadParameter { (onlyPayload) =>
-        val targetEntity = AgoraEntity(Option(namespace), Option(name), Option(snapshotId))
-
-        get { requestContext =>
-          completeWithPerRequest(requestContext, targetEntity, username, toBool(onlyPayload), path, queryHandlerProps)
-        } ~
-        delete { requestContext =>
-          completeEntityDelete(requestContext, targetEntity, username, path, queryHandlerProps)
-        } ~
-        post {
-          // only allow copying (post) for methods
-          if (path != AgoraConfig.methodsRoute) {
-            reject(MethodRejection(HttpMethods.GET), MethodRejection(HttpMethods.DELETE))
-          } else {
-            entity(as[AgoraEntity]) { newEntity =>
-              AgoraEntity.validate(newEntity) match {
-                case Success(_) => //noop
-                case Failure(errors) => throw new IllegalArgumentException(s"Method is invalid: Errors: $errors")
-              }
-              parameters("redact".as[Boolean] ? false) { redact =>
-                requestContext => completeEntityCopy(requestContext, targetEntity, newEntity, redact, username, path, queryHandlerProps)
+      extractOnlyPayloadParameter { (onlyPayload: Option[String]) =>
+        extractPayloadAsObjectParameter { (payloadAsObject: Option[String]) =>
```

Comment:
looks like these two could be simplified into a single spray directive, something like:
`parameters('onlyPayload, 'payloadAsObject) { (onlyPayload.as[Boolean], payloadAsObject.as[Boolean]) => `

and then remove the calls to `toBool` later. This would lose the error-handling/default in `toBool` but I'm okay with making the route stricter - if someone specifies something other than a boolean in these params, I think they *should* get an error instead of a default false/